sql/src/test/java/org/apache/druid/sql/calcite/CalciteQueryTest.java

```diff
@@ -17086,4 +17086,32 @@ public void testGroupingSetsWithLimitOrderByGran() throws Exception
         ).build()
     );
   }
+
+
+  @Test
+  public void testLookupWithNull() throws Exception
+  {
+    testQuery(
+        "SELECT dim2 ,lookup(dim2,'lookyloo') from foo where dim2 is null",
+        ImmutableList.of(
+            new Druids.ScanQueryBuilder()
+            .dataSource(CalciteTests.DATASOURCE1)
+            .intervals(querySegmentSpec(Filtration.eternity()))
+            .virtualColumns(
+                expressionVirtualColumn("v0", "null", ValueType.STRING)
+            )
+            .columns("v0")
+            .legacy(false)
+            .filters(new SelectorDimFilter("dim2", "", null))
+            .resultFormat(ScanQuery.ResultFormat.RESULT_FORMAT_COMPACTED_LIST)
+            .context(QUERY_CONTEXT_DEFAULT)
+            .build()
+        ),
+        ImmutableList.<Object[]>builder().add(
+            new Object[]{"", NULL_STRING},
+            new Object[]{"", NULL_STRING},
+            new Object[]{"", NULL_STRING}
```

Comment:
It looks like this is failing CI in SQL compatible null handling mode because the expected results are different, where there are only 2 null valued `dim2` rows instead of 3. The data for the 'foo' datasource is here: https://github.com/apache/druid/blob/master/sql/src/test/java/org/apache/druid/sql/calcite/util/CalciteTests.java#L368. In default mode (`druid.generic.useDefaultValueForNull=true`), the empty string is equivalent to null, so https://github.com/apache/druid/blob/master/sql/src/test/java/org/apache/druid/sql/calcite/util/CalciteTests.java#L390 is treated as null, but when `druid.generic.useDefaultValueForNull=false`, only https://github.com/apache/druid/blob/master/sql/src/test/java/org/apache/druid/sql/calcite/util/CalciteTests.java#L382 and https://github.com/apache/druid/blob/master/sql/src/test/java/org/apache/druid/sql/calcite/util/CalciteTests.java#L409 have null valued `dim2`. Additionally, `dim2` would be `null` instead of the empty string in the results.

Reply:
Thanks for reminding me. I have chang dimension `dim2` to `dim1` to solution this problem

Reply:
I restarted CI for the failing test. Looking at the new test though, I'm not sure it tests the issue anymore since the input to the lookup is no longer null? I actually think maybe the old test was better, you just needed to adjust the expected output based on whether or not it is in default null handling mode or not.e.g 
```
List<Object[]> expected;
if (useDefault) {
  expected = ImmutableList.<Object[]>builder().add(
...
```
like some of the other tests are doing.

Reply:
ok, I think you are right , we should consider specail  situation.   
 I hava changed it to old test, and handled `null` case.